### ISSUE TITLE
Feature Request: 4:4:4 Chroma Subsampling Option

### ISSUE THREAD

**Issue by temp-64GTX** [gopro/labs#84]
It's not a secret that Gopro makes video in 4:2:0 color format. Which is sad for hi-res color grading in post-production.
So. You make the possibility for bitrate over 100Mb/s. Maybe there is also possible to make the option for 4:4:4 color? 

...Thanks for the labs, very useful stuff.

![4-2-0](https://user-images.githubusercontent.com/107313519/189548526-b6a25f42-0796-491e-85d0-710d11a11980.PNG)

4:4:4 makes world happy

![4-4-4](https://user-images.githubusercontent.com/107313519/189548532-ab3b1f3f-13b9-48a2-bc10-2b36926695ea.PNG)

![chroma](https://user-images.githubusercontent.com/107313519/189548584-d10e6bba-80cc-4503-b03d-26c4a1de2ecb.PNG)

**Comment by dnewman-gpsw** [comment]
That will not be possible, and the hardware encoder doesn't support it, and I expect many HEVC hardware decoders wouldn't either.  Finally a CFA Bayer sensor (most cameras these days) can't resolve 4:4:4 at the native resolution (not even 4:2:2 is practical), so there wouldn't be a significant improvement (see https://en.wikipedia.org/wiki/Bayer_filter.)  If you take a 4:2:0 4K image you can make a 4:4:4 HD image.  So if you only need HD, you can do that.  From a HERO10, at 5.3K 4:3, the highest 4:4:4 resolution would be 2656x1992.

**Comment by temp-64GTX** [comment]
That's extremely sad. Extremely :(

**Comment by dnewman-gpsw** [comment]
Not really, all modern cameras have this limitation.  It is the nature of the bayer sensor.  True 4:4:4 was only available for 3 chip cameras or something like Thompson Viper (but only 444 in HD.)  While some very high-end cameras may support encoding ProRES 4444, the true source chroma resolution is lower.

**Comment by temp-64GTX** [comment]
I'm working with 3D-graphics, and donno much about chip cameras. But when i render frame in some 3d application - it's always have true 32-bit depth, and no subsampling at all. Very handy and useful. Much of things can be done in post-effects. You can turn night picture to a bright-sunny-day picture. And day to night. Very conveniently. 

Once upon a time i noticed that my final output video have some poor faded colors in small details. Then i learn about this nasty 4:2:0. Before that - i didn't dive deep to highly specialized video-compression things. I was sure that videos is always had full color-depth. But they are not. From that time - i always make my output videos in 4:4:4 format.

Anyway, thanks for information. Crossing fingers that gopro will implement that possibility in future cameras.

**Comment by dnewman-gpsw** [comment]
In post you can synthetically up convert 4:2:0 to 4:4:4, you shouldn't need to do that, however it seems your workflow is poorly handling sub-sampled chroma. Look into that, as it will be easier to fix than waiting for 4:4:4 from any single sensor camera (see explanation above.)
